### PR TITLE
CANDLEPIN-619: Re-run PR workflow when target branch gets pushed

### PR DESCRIPTION
**Pull request by bakajstep**
- Create a workflow that triggers when a PR is closed, and it will rerun all workflows associated with the affected branch

### CODE REVIEW

File: .github/workflows/rerun_workflow_on_merge.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  rerun-workflows:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Trigger workflow runs
+        run: |
+          token="${{ secrets.GITHUB_TOKEN }}"
```
Comment (JoshAlbrecht23) [inline .github/workflows/rerun_workflow_on_merge.yml:18]: I think we should mask this so that it doesn't potentially leak in the logs

**Review by JoshAlbrecht23 — CHANGES_REQUESTED** [review]

File: .github/workflows/rerun_workflow_on_merge.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Rerun Workflows on PR Merge
```
[the remark this answers is not in the document]
Comment (bakajstep) [reply .github/workflows/rerun_workflow_on_merge.yml]: that's a good idea. Done

File: .github/workflows/rerun_workflow_on_merge.yml (hunk 1 above)
Comment (bakajstep) [reply .github/workflows/rerun_workflow_on_merge.yml]: done

**Comment by bakajstep** [comment]
We are not able to test this until we merge it to main so i write script to test it. So whoever's going to go through it can give it a try.
```shell
#!/bin/bash

pr_head_shas=$(gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/candlepin/candlepin/pulls | jq -r '.[] | select(.base.ref == "main") | .head.sha')

for head_sha in $pr_head_shas; do
    runner_id=$(gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/candlepin/candlepin/actions/workflows/pr_verification.yml/runs?head_sha=$head_sha | jq -r '.workflow_runs[] | .id')
    workflow_ids+=($runner_id)
done


for id in "${workflow_ids[@]}"
do
    gh api \
  --method POST \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  /repos/candlepin/candlepin/actions/runs/$id/rerun
done
``` 

You must be logged with account in candlepin organization to make request the gh api of the candlepin organization.
[https://cli.github.com/manual/gh_auth_login](url)

**Comment by nikosmoum** [comment]
We need to discuss an alternative to this during retro

**Comment by nikosmoum** [comment]
Closing this in favor of using GH Merge Queues and adding a step to merge the target branch in the feature branch before running the CI checks

**Pull request closed without merging**